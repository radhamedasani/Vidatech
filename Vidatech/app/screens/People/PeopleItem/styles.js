import { StyleSheet } from 'react-native'
import { padding, colors, fontSize } from '../../../themes/base'

export const peopleItemStyle = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        marginBottom: padding.sm,
        marginHorizontal: padding.sm,
        paddingLeft: padding.sm,
        backgroundColor: colors.secondary
    },
    avatarStyle: {
        width: 70,
        height: 70,
    },
    nameStyle: {
        fontSize: fontSize.md
    },
    details: {
        paddingVertical: padding.sm,
        paddingLeft: padding.md
    },
    designation: {
        color: colors.teritary,
        fontSize: fontSize.sm
    },
    emailTxt: {
        color: 'blue',
        fontSize: fontSize.sm
    }
})