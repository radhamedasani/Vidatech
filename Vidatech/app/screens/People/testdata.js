
export const people = {
    "id": "1",
    "createdAt": "2019-04-29T10:04:24.713Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/johndezember/128.jpg",
    "jobTitle": "Dynamic Implementation Designer",
    "phone": "556.662.4422 x3344",
    "favouriteColor": "#023a77",
    "email": "Benny78@gmail.com",
    "firstName": "Jan",
    "lastName": "Thompson",
    "name": "Jan Thompson",
}

export const PeopleResponse = [{
    "id": "1",
    "createdAt": "2019-04-29T10:04:24.713Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/johndezember/128.jpg",
    "jobTitle": "Dynamic Implementation Designer",
    "phone": "556.662.4422 x3344",
    "favouriteColor": "#023a77",
    "email": "Benny78@gmail.com",
    "firstName": "Jan",
    "lastName": "Thompson"
}, {
    "id": "2",
    "createdAt": "2019-04-28T21:05:15.710Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/lingeswaran/128.jpg",
    "jobTitle": "Lead Response Coordinator",
    "phone": "626-316-8058 x077",
    "favouriteColor": "#39653b",
    "email": "Jordon.Parker30@hotmail.com",
    "firstName": "Rhett",
    "lastName": "Carter"
}, {
    "id": "3",
    "createdAt": "2019-04-29T10:33:47.141Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/ruehldesign/128.jpg",
    "jobTitle": "Lead Data Planner",
    "phone": "1-852-834-5571 x8035",
    "favouriteColor": "#1b366f",
    "email": "Marcus.Kuphal78@hotmail.com",
    "firstName": "Rodrigo",
    "lastName": "O'Conner"
}, {
    "id": "4",
    "createdAt": "2019-04-28T20:35:29.203Z",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/kolsvein/128.jpg",
    "jobTitle": "Product Factors Developer",
    "phone": "646.998.1257",
    "favouriteColor": "#6e0a75",
    "email": "Allene.Keeling@hotmail.com",
    "firstName": "Jamey",
    "lastName": "Mayer"
}]