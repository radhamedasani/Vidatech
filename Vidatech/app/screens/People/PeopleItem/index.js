import React, { Component } from 'react'
import { View, Text, Image, ActivityIndicator } from 'react-native'
import { peopleItemStyle } from './styles'
import { formatDate } from '../../../utilities'
import { ImageLoader } from '../../../components/ImageLoader'
import { BoxContainer } from '../../../components/BoxContainer'

export function PeopleItem(props) {
    const { avatar, name, email, jobTitle, favouriteColor, createdAt } = props;
    const { details, nameStyle, emailTxt, designation, avatarStyle } = peopleItemStyle;
    return (<BoxContainer>
        <ImageLoader avatarStyle={avatarStyle} avatar={avatar} />
        <View style={details}>
            <Text style={[nameStyle, { color: favouriteColor }]}>{`${name}`}</Text>
            <Text style={designation}>{jobTitle}</Text>
            <Text style={designation}>{`Joined On ${formatDate(createdAt)}`}</Text>
            <Text style={emailTxt}>{email}</Text>
        </View>
    </BoxContainer>)
}



