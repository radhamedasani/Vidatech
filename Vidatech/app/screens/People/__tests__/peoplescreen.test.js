import React from 'react'
import reactTestRenderer from 'react-test-renderer'

import { PeopleItem } from '../PeopleItem'
import { People } from '../'
import { people, PeopleResponse } from '../../testdata'

describe('People Screen Tests ==> ', () => {

    it('Test People Item', () => {
        const peopleitemComp = reactTestRenderer.create(<PeopleItem {...people} />);
        expect(peopleitemComp).toMatchSnapshot();
    })

    it('Test People Comp', () => {
        global.fetch = jest.fn(() => {
            json: () => PeopleResponse
        })
        const PeopleComp = reactTestRenderer.create(<People />);
        expect(PeopleComp).toMatchSnapshot();
    })
})