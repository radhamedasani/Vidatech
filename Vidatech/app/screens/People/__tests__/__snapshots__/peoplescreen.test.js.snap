// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Screen Tests ==>  Test People Comp 1`] = `
<View
  style={
    Object {
      "backgroundColor": "lightgrey",
      "flex": 1,
      "marginTop": 60,
      "paddingBottom": 60,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#882758",
        "height": 60,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
        }
      }
    >
      Vidatec
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 10,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;

exports[`People Screen Tests ==>  Test People Item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "row",
      "marginBottom": 10,
      "marginHorizontal": 10,
      "paddingLeft": 10,
    }
  }
>
  <View
    style={
      Object {
        "height": 70,
        "width": 70,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "position": "absolute",
          },
          Object {
            "height": 70,
            "width": 70,
          },
        ]
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "position": "absolute",
          },
          Object {
            "height": 70,
            "width": 70,
          },
        ]
      }
    >
      <Image
        onError={[Function]}
        onLoad={[Function]}
        source={
          Object {
            "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/johndezember/128.jpg",
          }
        }
        style={
          Object {
            "height": 70,
            "width": 70,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "paddingLeft": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "#023a77",
          },
        ]
      }
    >
      Jan Thompson
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 14,
        }
      }
    >
      Dynamic Implementation Designer
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 14,
        }
      }
    >
      Joined On 29 Apr 2019
    </Text>
    <Text
      style={
        Object {
          "color": "blue",
          "fontSize": 14,
        }
      }
    >
      Benny78@gmail.com
    </Text>
  </View>
</View>
`;
